agent/exec/controller.go

```diff
@@ -172,21 +191,37 @@ func Do(ctx context.Context, task *api.Task, ctlr Controller) (*api.TaskStatus,
 			return
 		}
 
-		cctlr, ok := ctlr.(ContainerController)
-		if !ok {
-			return
+		if containerStatus == nil {
+			// collect this, if we haven't
+			cctlr, ok := ctlr.(ContainerStatuser)
+			if !ok {
+				return
+			}
+
+			var err error
+			containerStatus, err = cctlr.ContainerStatus(ctx)
+			if err != nil {
+				log.G(ctx).WithError(err).Error("container status unavailable")
+			}
+
+			// at this point, things have gone fairly wrong. Remain positive
```

Comment:
![Be an optimist prime](http://img05.deviantart.net/068c/i/2010/247/f/4/optimist_prime_by_avid-d2xz9e1.jpg)